src/main/java/de/tum/in/www1/artemis/service/scheduled/ProgrammingExerciseScheduleService.java

```diff
@@ -140,7 +140,11 @@ private static boolean needsToBeScheduled(ProgrammingExercise exercise) {
             return true;
         }
         // Manual assessed programming exercises as well
-        if (exercise.getAssessmentType() != AssessmentType.AUTOMATIC) {
+        if (AssessmentType.AUTOMATIC != exercise.getAssessmentType()) {
+            return true;
+        }
+        // Exercises where students can complain as well
+        if (exercise.getAllowComplaintsForAutomaticAssessments()) {
             return true;
         }
```

Comment:
Is there a reason why these statements are not combined? 
```suggestion
        // Manual assessed programming and exercises where students can complain as well
        if (AssessmentType.AUTOMATIC != exercise.getAssessmentType() || exercise.getAllowComplaintsForAutomaticAssessments()) {
            return true;
        }
```
_GitHub didn't let me mark the comment above aswell_

Reply:
The whole method was written in that style. Since the method has like 5 to 6 of these statements, a combination of all would probably not be easy to understand.